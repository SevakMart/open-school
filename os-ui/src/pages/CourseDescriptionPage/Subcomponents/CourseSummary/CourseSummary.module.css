.mainContent {
  border-radius: 40px;
  align-self: flex-start;
  border: 2px solid black;
}
.mainContent .headerContent {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  background-color: #F9F9FD;
  border-radius: 40px 40px 0px 0px;
}
.mainContent .headerContent h2 {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 600;
  font-size: 20px;
  line-height: 24px;
  color: #333941;
  margin: 32px 162px 32px 48px;
}
.mainContent .headerContent .headerIcons {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin-right: 83px;
}
.mainContent .headerContent .headerIcons img {
  margin-right: 32px;
}
.mainContent .courseSummaryItemList {
  margin-top: 48px;
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  gap: 24px 0;
  justify-content: none;
}
.mainContent .userEnrollText {
  margin: 66px auto 24px;
  text-align: center;
}
.mainContent .buttonContainer {
  margin: 64px 60px 80px 48px;
}
.mainContent .enrolledButtonContainer {
  margin: 24px 60px 80px 48px;
}/*# sourceMappingURL=CourseSummary.module.css.map */