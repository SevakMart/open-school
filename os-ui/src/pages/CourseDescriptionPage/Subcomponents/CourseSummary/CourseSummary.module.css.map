{"version":3,"sources":["CourseSummary.module.scss","CourseSummary.module.css","../../../../styles/Mixins/FlexMixins.scss","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss","../../../../styles/Mixins/GridMixin.scss"],"names":[],"mappings":"AAOA;EACI,mBAAA;EACA,sBAAA;EACA,uBAAA;ACNJ;ADOI;EEVF,aAAA;EACA,mBFUyB;EETzB,mBFS6B;EER7B,8BFQoC;EAC9B,yBAAA;EACA,gCAAA;ACFR;ADGQ;EGdN,yBCDY;EDEZ,kBHc2C;EGb3C,gBHakD;EGZlD,eHYsD;EGXtD,iBHW2D;EACjD,cKZO;ELaP,4BAAA;ACGZ;ADDQ;EEnBN,aAAA;EACA,mBFmB6B;EElB7B,mBFkBkC;EEjBlC,8BFiB0C;EAChC,kBAAA;ACMZ;ADLY;EACI,kBAAA;ACOhB;ADHI;EACI,gBAAA;EM5BJ,aAAA;EACA,qCN4BuB;EM3BvB,WAAA;EACA,qBN0BqC;ACQzC;ADNI;EACI,sBAAA;EACA,kBAAA;ACQR;ADNI;EACI,2BAAA;ACQR;ADNI;EACI,2BAAA;ACQR","file":"CourseSummary.module.css"}