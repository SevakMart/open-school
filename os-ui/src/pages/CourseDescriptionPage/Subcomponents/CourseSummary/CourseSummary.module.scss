@import "../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../styles/Mixins/FontMixin.scss";
@import "../../../../styles/Mixins/ButtonMixin.scss";
@import "../../../../styles/Configs/Colors.scss";
@import "../../../../styles/Configs/Fonts.scss";
@import "../../../../styles/Mixins/GridMixin.scss";

.mainContent {
    border-radius: 40px;
    align-self: flex-start;
    border: 2px solid black;
    .headerContent{
        @include flexStyle(row,center,space-between);
        background-color: #F9F9FD;
        border-radius: 40px 40px 0px 0px;
        h2{
            @include fontStyle($primary-font,normal,600,20px,24px);
            color: $black-text-color;
            margin:32px 162px 32px 48px;
        }
        .headerIcons {
            @include flexStyle(row, center, space-between);
            margin-right: 83px;
            img {
                margin-right:32px;
            }
        }
    }
    .courseSummaryItemList{
        margin-top:48px;
        @include gridStyle(repeat(1,1fr),none,24px,0);
    }
    .userEnrollText {
        margin:66px auto 24px;
        text-align: center;
    }
    .buttonContainer {
        margin:64px 60px 80px 48px;
    }
    .enrolledButtonContainer {
        margin:24px 60px 80px 48px;
    }
}
