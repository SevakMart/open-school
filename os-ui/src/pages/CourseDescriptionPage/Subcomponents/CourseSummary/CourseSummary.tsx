import { useTranslation } from 'react-i18next';
import { CourseDescriptionType } from '../../../../types/CourseDescriptionType';
import BookmarkIcon from '../../../../icons/Bookmark';
import ShareIcon from '../../../../icons/Share';
import CourseSummaryItem from './Subcomponent/CourseSummaryItem/CourseSummaryItem';
import styles from './CourseSummary.module.scss';

const CourseSummary = ({
  rating, enrolled, level, language, duration,
}:Omit<CourseDescriptionType, 'title'|'description'|'goal'|'modules'|'mentorDto'>) => {
  const { t } = useTranslation();
  const courseSummaryItem = {
    rating, enrolled, level, language, duration,
  };
  const {
    mainContent, headerContent, headerIcons, courseSummaryItemList,
  } = styles;
  return (
    <div className={mainContent}>
      <div className={headerContent}>
        <h2>{t('Course Summary')}</h2>
        <div className={headerIcons}>
          <ShareIcon iconSize="0.5 rem" />
          <BookmarkIcon iconSize="0.5 rem" />
        </div>
      </div>
      <div className={courseSummaryItemList}>
        {
          Object.entries(courseSummaryItem).map((item) => (
            <CourseSummaryItem
              key={item[0]}
              title={item[0]}
              value={item[1]}
            />
          ))
        }
      </div>
      <button type="button">{t('Enroll')}</button>
    </div>
  );
};
export default CourseSummary;
