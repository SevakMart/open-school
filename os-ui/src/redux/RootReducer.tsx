import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
