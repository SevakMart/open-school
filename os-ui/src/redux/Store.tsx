import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
