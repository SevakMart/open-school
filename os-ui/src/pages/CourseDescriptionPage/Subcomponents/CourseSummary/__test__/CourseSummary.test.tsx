import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { PayloadAction } from '@reduxjs/toolkit';
import { store } from '../../../../../redux/Store';
import CourseSummary from '../CourseSummary';

jest.mock('redux-state-sync', () => ({
  createStateSyncMiddleware:
    () => () => (next: (action: PayloadAction) => void) => (action: PayloadAction) => next(action),
  initMessageListener: () => jest.fn(),
}));

describe('Create test cases for course description page', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <CourseSummary
          rating={4.5}
          enrolled={2}
          level="Advanced"
          language="French"
          duration={360}
          enrollInCourse={() => null}
          isEnrolled={false}
          courseId={2}
          userIdAndToken={{ id: 2, token: 'jksndfkjnskfk' }}
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
