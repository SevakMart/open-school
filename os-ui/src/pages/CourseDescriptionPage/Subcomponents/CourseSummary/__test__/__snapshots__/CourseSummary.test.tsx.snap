// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for course description page Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContent"
  >
    <div
      class="headerContent"
    >
      <h2>
        string.courseDescription.title.summary
      </h2>
      <div
        class="headerIcons"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="18"
            cy="5"
            r="3"
          />
          <circle
            cx="6"
            cy="12"
            r="3"
          />
          <circle
            cx="18"
            cy="19"
            r="3"
          />
          <line
            x1="8.59"
            x2="15.42"
            y1="13.51"
            y2="17.49"
          />
          <line
            x1="15.41"
            x2="8.59"
            y1="6.51"
            y2="10.49"
          />
        </svg>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(94, 97, 123); cursor: pointer; fill: none;"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21l-7-5-7 5V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2z"
          />
        </svg>
      </div>
    </div>
    <div
      class="courseSummaryItemList"
    >
      <div
        class="mainContainer"
      >
        <div
          class="mainSubContainer"
        >
          <div
            class="dotNotation"
          />
          <div
            class="mainContent"
          >
            <h2>
              rating
            </h2>
            <p>
              4.5
            </p>
          </div>
        </div>
      </div>
      <div
        class="mainContainer"
      >
        <div
          class="mainSubContainer"
        >
          <div
            class="dotNotation"
          />
          <div
            class="mainContent"
          >
            <h2>
              enrolled
            </h2>
            <p>
              2
            </p>
          </div>
        </div>
      </div>
      <div
        class="mainContainer"
      >
        <div
          class="mainSubContainer"
        >
          <div
            class="dotNotation"
          />
          <div
            class="mainContent"
          >
            <h2>
              level
            </h2>
            <p>
              Advanced
            </p>
          </div>
        </div>
      </div>
      <div
        class="mainContainer"
      >
        <div
          class="mainSubContainer"
        >
          <div
            class="dotNotation"
          />
          <div
            class="mainContent"
          >
            <h2>
              language
            </h2>
            <p>
              French
            </p>
          </div>
        </div>
      </div>
      <div
        class="mainContainer"
      >
        <div
          class="mainSubContainer"
        >
          <div
            class="dotNotation"
          />
          <div
            class="mainContent"
          >
            <h2>
              duration
            </h2>
            <p>
              360
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="buttonContainer"
    >
      <button
        type="button"
      >
        button.enroll
      </button>
    </div>
  </div>
</DocumentFragment>
`;
